spec/std/big/big_decimal_spec.cr

```diff
@@ -0,0 +1,223 @@
+    BigDecimal.new(0).should eq(BigDecimal.new(1) * BigDecimal.new(0))
+    BigDecimal.new(-1).should eq(BigDecimal.new(1) * BigDecimal.new(-1))
+    BigDecimal.new(-1).should eq(BigDecimal.new(-1) * BigDecimal.new(1))
+    BigDecimal.new("1.2621466432").should eq(BigDecimal.new("1.12345") * BigDecimal.new("1.123456"))
+    BigDecimal.new("1.2621466432").should eq(BigDecimal.new("1.123456") * BigDecimal.new("1.12345"))
+
+    expect_raises(DivisionByZero) do
+      BigDecimal.new(0) / BigDecimal.new(0)
+    end
+    expect_raises(DivisionByZero) do
+      BigDecimal.new(1) / BigDecimal.new(0)
+    end
+    expect_raises(DivisionByZero) do
+      BigDecimal.new(-1) / BigDecimal.new(0)
+    end
+    BigDecimal.new(1).should eq(BigDecimal.new(1) / BigDecimal.new(1))
+    BigDecimal.new("0.5").should eq(BigDecimal.new(1) / BigDecimal.new(2))
+    BigDecimal.new("-0.5").should eq(BigDecimal.new(1) / BigDecimal.new(-2))
+    BigDecimal.new(0).should eq(BigDecimal.new(0) / BigDecimal.new(1))
+
+    BigDecimal.set_max_div_iterations(5)
+    BigDecimal.new(33333.to_big_i, 5.to_u64).should eq(BigDecimal.new(1) / BigDecimal.new(3))
+    BigDecimal.new(33.to_big_i, 5.to_u64).should eq(BigDecimal.new(1) / BigDecimal.new(3000))
+
+    BigDecimal.set_max_div_iterations(7)
+    BigDecimal.new(3333333.to_big_i, 7.to_u64).should eq(BigDecimal.new(1) / BigDecimal.new(3))
+    BigDecimal.new(3333.to_big_i, 7.to_u64).should eq(BigDecimal.new(1) / BigDecimal.new(3000))
+  end
+
+  it "can be converted from other types" do
+    1.to_big_d.should eq (BigDecimal.new(1))
+    "1.5".to_big_d.should eq (BigDecimal.new("1.5"))
+    BigInt.new(15).to_big_d.should eq (BigDecimal.new("15"))
+  end
+
+  it "is comparable with other types" do
+    BigDecimal.new("1.0").should eq BigDecimal.new("1")
+    BigDecimal.new("1").should eq BigDecimal.new("1.0")
+    BigDecimal.new(1, 10).should eq BigDecimal.new(10, 11)
+    BigDecimal.new(10, 11).should eq BigDecimal.new(1, 10)
+
+    (BigDecimal.new(1) > BigDecimal.new(1)).should be_false
+    (BigDecimal.new("1.00000000000000000000000000000000000001") > BigDecimal.new(1)).should be_true
+    (BigDecimal.new("0.99999999999999999999999999999999999999") > BigDecimal.new(1)).should be_false
+    BigDecimal.new("1.00000000000000000000000000000000000000").should eq BigDecimal.new(1)
+
+    (1 < BigDecimal.new(1)).should be_false
+
+    (1 < BigDecimal.new(1)).should be_false
+    (BigDecimal.new(1) < 1).should be_false
+    (2 < BigDecimal.new(1)).should be_false
+    (BigDecimal.new(2) < 1).should be_false
+    (BigDecimal.new("-1") > BigDecimal.new("1")).should be_false
+
+    (1 > BigDecimal.new(1)).should be_false
+    (BigDecimal.new(1) > 1).should be_false
+    (2 > BigDecimal.new(1)).should be_true
+    (BigDecimal.new(2) > 1).should be_true
+    (BigDecimal.new("-1") < BigDecimal.new("1")).should be_true
+
+    (1 >= BigDecimal.new(1)).should be_true
+  end
+
+  it "keeps precision" do
+    oneThousandth = BigDecimal.new("0.001")
```

Comment:
please, use `snake_case` for variable names; we ain't in JavaScript Land Dorothy ;)

Reply:
Heh. Fixed.